### PR TITLE
325 improve testsuite simplify testing

### PR DESCRIPTION
**Pull request by carakas**
## Type

- Enhancement

## Resolves the following issues

Based on #3025 

## Pull request description

Add some extra test methods to make it easier to write tests in Fork and easier to read the tests
Some cleanup and some Data fixtures

All the tests are now running in a sandbox so it takes a little longer but the issue with backend constants being available in frontend tests is solved now.

struggled a bit fixing the coverage reports in scrutinizer but the coverage should be more accurate now and it should fluctuate less

### CODE REVIEW

**Comment by carakas** [comment]
struggled a bit fixing the coverage reports in scrutinizer but the coverage should be more accurate now and it should fluctate less

File: src/ForkCMS/Bundle/InstallerBundle/Tests/Controller/InstallerControllerTest.php [hunk 1]
```diff
@@ -6,12 +6,31 @@
...
+    private $kernelDir;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->kernelDir = $this->getProvidedData()[0]->getContainer()->getParameter('kernel.project_dir') . '/app';
+    }
+
+    protected function onNotSuccessfulTest($e): void
```
Comment (bjorvack) [inline src/ForkCMS/Bundle/InstallerBundle/Tests/Controller/InstallerControllerTest.php]: Use a better variable name

File: src/Backend/Modules/Blog/Tests/Actions/EditTest.php [hunk 2]
```diff
@@ -28,11 +28,7 @@ public function testWeCanGoToEditFromTheIndexPage(Client $client): void
         $this->login($client);
 
         $this->assertPageLoadedCorrectly($client, '/private/en/blog/index', [LoadBlogPosts::BLOG_POST_TITLE]);
-
-        $link = $client->getCrawler()->selectLink(LoadBlogPosts::BLOG_POST_TITLE)->link();
-        $client->click($link);
-
-        $this->assertIs200($client);
+        $this->assertClickOnLink($client, LoadBlogPosts::BLOG_POST_TITLE, [LoadBlogPosts::BLOG_POST_TITLE]);
         $this->assertCurrentUrlContains($client, '&id=1');
```
Comment (bjorvack) [inline src/Backend/Modules/Blog/Tests/Actions/EditTest.php]: Use a constant for the id like the other tests

File: src/Backend/Modules/Tags/Tests/Actions/EditTest.php [hunk 3]
```diff
@@ -25,112 +24,85 @@ protected function setUp(): void
...
-        $client->click($link);
-
-        self::assertEquals(200, $client->getResponse()->getStatusCode());
-        self::assertContains(
-            '&id=2',
-            $client->getHistory()->current()->getUri()
-        );
+        $this->assertPageLoadedCorrectly($client, '/private/en/tags/index', [LoadTagsTags::TAGS_TAG_2_NAME]);
+        $this->assertClickOnLink($client, LoadTagsTags::TAGS_TAG_2_NAME, [LoadTagsTags::TAGS_TAG_2_NAME]);
+        $this->assertCurrentUrlContains($client, '&id=2');
```
[the remark this answers is not in the document]
Comment (carakas) [reply src/Backend/Modules/Tags/Tests/Actions/EditTest.php]: Fixed in 3782ab06b

File: src/ForkCMS/Bundle/InstallerBundle/Tests/Controller/InstallerControllerTest.php (hunk 1 above)
Comment (carakas) [reply src/ForkCMS/Bundle/InstallerBundle/Tests/Controller/InstallerControllerTest.php]: This is a method from phpunit and part of their interface so I can't change it

File: src/Backend/Modules/Blog/Tests/Actions/EditTest.php (hunk 2 above)
Comment (carakas) [reply src/Backend/Modules/Blog/Tests/Actions/EditTest.php]: Fixed in ec7b66c48

File: src/ForkCMS/Bundle/InstallerBundle/Tests/Controller/InstallerControllerTest.php (hunk 1 above)
Comment (carakas) [reply src/ForkCMS/Bundle/InstallerBundle/Tests/Controller/InstallerControllerTest.php]: fixed in 0cf43e9 and 93189de

**Pull request merged**